### DISCUSSION TITLE
Read a Flow of Debug

### DISCUSSION THREAD

**Discussion by bgile** [nornir-automation/nornir#747] [General]
Hey all, looking for a way to read a stream of debug and then capture it. I have to use netmiko in particular for this use case.  I may be wrong, but, netmiko is built on finding the prompt, and in my case, I don't have a prompt, I just want to read the channel for "X" amount of time and then capture.  Is anyone familiar with doing something like this?

Thanks

**Comment by ktbyers** [comment]
Netmiko has a read_channel() method which will read all the available data (at that point in time). So you can build a loop and sleep and keep reading using read_channel() for as long as you want.

You can get the underlying Netmiko connection inside of a Nornir custom task.

**Reply by bgile** [reply]
Perfect, thanks Kirk.  I totally forgot about read_channel.  Doh!  I take it something like this should suffice to get me the underlying Netmiko connection object?
```
netmiko_conn = task.host.get_connection("netmiko", configuration=task.nornir.config)
```

**Comment by bgile** [comment]
Small inspection of the object:
```
netmiko_conn = task.host.get_connection("netmiko", configuration=task.nornir.config).connection
```
Gets me access to read_channel().
```
